### PR TITLE
Feature: create navbar 

### PR DESCRIPTION
**Pull request by teoh4770**
Fix: https://github.com/LaurierHawkHacks/Landing/issues/91

### What is this pr do?

Setup navbar and menu, and make it responsive

### Things to implement in the future

- [ ] link styling
- [ ] replace placeholder links with actual links
- [ ] navigate to different section of landing page
- [ ] navigate to portal
- [X] background color
- [X] remove banner shadow

### Preview:

<figure>
	Screenshot 2024-02-12 at 5 56 48 PM
	<figcaption>Mobile View</figcaption>
</figure>
<br>

<figure>
	Screenshot 2024-02-12 at 5 58 26 PM
	<figcaption>Menu</figcaption>
</figure>
<br>

<figure>
	Screenshot 2024-02-12 at 5 59 17 PM
	<figcaption>Desktop View</figcaption>
</figure/>
<br>

### CODE REVIEW

File: src/components/Navbar/NavItems.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}> = ({ isHorizontal, handleClick }) => {
+    const rowStyle = 'flex-row items-center';
+    const colStyle = 'flex-col lg:p-0';
+    const scrollPadding = -150;
+
+    return (
+        <ul className={`flex gap-7 ${isHorizontal ? rowStyle : colStyle}`}>
+            {links.map((link, index) => (
+                <li key={index}>
+                    <Link
```
Comment (juancwu) [inline src/components/Navbar/NavItems.tsx:24]: Links should be of the green colour like they are in Figma.

File: src/components/Navbar/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                <NavItems isHorizontal={true} handleClick={hideMenu} />
+            </div>
+
+            {/* portal button */}
+            <div
+                className={`portal-btn hidden lg:block transition-all duration-500 ease-in-out ${
+                    scrollPosY < 100 ? 'lg:mr-32' : 'lg:mr-0'
+                }`}
+            >
+                <button className="px-4 py-2 xl:px-6 xl:py-3 rounded-lg">Application Portal</button>
```
[the remark this answers is not in the document]
Comment (teoh4770) [reply src/components/Navbar/Navbar.tsx:53]: @juancwu I will let you handle the button component :>

File: src/components/Navbar/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    size={25}
+                    label="Show menu"
+                    toggled={showMenu}
+                    toggle={() => setShowMenu(!showMenu)}
+                />
+            </div>
+
+            {/* side menu */}
+            <div className="side-menu absolute">
+                {showMenu && <Menu showMenu hideMenu={hideMenu} />}
```
Comment (juancwu) [inline src/components/Navbar/Navbar.tsx:94]:
```suggestion
                <Menu showMenu={showMenu} hideMenu={hideMenu} />
```

**Comment by teoh4770** [comment]
let me know if the body lock screen code implementation is good :>

**Review by juancwu — APPROVED** [review]

**Pull request merged**